jetty-io/src/main/java/org/eclipse/jetty/io/RetainableByteBufferPool.java

```diff
@@ -55,4 +60,56 @@ static RetainableByteBufferPool findOrAdapt(Container container, ByteBufferPool
         }
         return retainableByteBufferPool;
     }
+
+    /**
+     * A Pool wrapper that collects sample statistics on the acquired
+     * buffers.
+     * @see SampleStatistic
+     */
+    @ManagedObject
+    class SampledPool implements RetainableByteBufferPool, Dumpable
```

Comment:
This class is implicitly `public`, which I don't think it's a good idea.

Reply:
We want at least this one to be public, so we can run real deployments and capture data about the pool usage.   You can't simultaneously argue that `ExponentialPool` is not justified by any data and then prevent anything that allows the data to be collected!